Tryouts/Plugins/ApplicationPlugins/process explorer/backend/ModuleProcessMonitor/Processes/ProcessInfoGeneratorWindows.cs

```diff
@@ -0,0 +1,360 @@
+        try
+        {
+            mo.Get();
+            ppid = Convert.ToInt32(mo["ParentProcessId"]);
+        }
+        catch (Exception exception)
+        {
+            if (process.Id > 0)
+            {
+                _logger.ManagementObjectPpid(process.Id, exception);
+            }
+        }
+
+        return ppid;
+
+    }
+
+    public override float GetMemoryUsage(Process process)
+    {
+        int memsize;
+
+        var perf = new PerformanceCounter()
+        {
+            CategoryName = "Process",
+            CounterName = "Working Set - Private",
+            InstanceName = process.ProcessName,
+        };
+
+        var memoryPerformanceCounter = _memoryPerformanceCounters.GetOrAdd(process.Id, perf);
+
+        memsize = Convert.ToInt32(memoryPerformanceCounter.NextValue()) / Convert.ToInt32(1024) / Convert.ToInt32(1024);
+
+        return (float)((memsize / GetTotalMemoryInMb()) * 100);
+    }
+
+    private static double GetTotalMemoryInMb()
+    {
+        var gcMemoryInfo = GC.GetGCMemoryInfo();
+        var installedMemory = gcMemoryInfo.TotalAvailableMemoryBytes;
+        return Convert.ToDouble(installedMemory) / 1048576.0;
+    }
+
+    public override float GetCpuUsage(Process process)
+    {
+        var performanceCounter = new PerformanceCounter(
+                "Process",
+                "% Processor Time",
+                process.ProcessName,
+                true);
+
+        var cpuPerformanceCounter = _memoryPerformanceCounters.GetOrAdd(process.Id, performanceCounter);
+
+        var processCpuUsage = cpuPerformanceCounter.NextValue();
+
+        return processCpuUsage / Environment.ProcessorCount;
+    }
+
+    public virtual SynchronizedCollection<ProcessInfoData> GetChildProcesses(Process process)
+    {
+        var children = new SynchronizedCollection<ProcessInfoData>();
+
+        ManagementObjectSearcher mos = new ManagementObjectSearcher(
+            string.Format(
+                "Select * From Win32_Process Where ParentProcessID={0} Or ProcessID={0}",
+                process.Id));
+
+        foreach (var objectCollection in mos.Get())
+        {
+            var managementObject = (ManagementObject)objectCollection;
+            try
+            {
+                var childProcess = Process.GetProcessById(Convert.ToInt32(managementObject["ProcessID"]));
+
+                var childProcessInfo = new ProcessInformation(childProcess);
+
+                ProcessInformation.SetProcessInfoData(childProcessInfo, this);
+
+                children.Add(childProcessInfo.ProcessInfo);
+            }
+            catch (Exception exception)
+            {
+                _logger.ManagementObjectChildError(exception);
+            }
+        }
+
+        return children;
+    }
+
+    private static SynchronizedCollection<int> GetProcessIds(IEnumerable<ProcessInfoData> processes)
+    {
+        var processIds = new SynchronizedCollection<int>();
```

Comment:
The collection returned from this method never seems to be used from multiple threads, consider using a simple list